front/components/editor/extensions/agent_builder/InstructionBlockExtension.tsx

```diff
@@ -145,24 +141,10 @@ const InstructionBlockComponent: React.FC<NodeViewProps> = ({
   updateAttributes,
 }) => {
   const [isCollapsed, setIsCollapsed] = useState(
-    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-    node.attrs.isCollapsed || false
+    node.attrs.isCollapsed ?? false
   );
 
-  let displayType = "";
-  if (node.childCount > 0) {
-    const firstChild = node.child(0);
-    if (firstChild.type.name === "paragraph") {
-      const text = firstChild.textContent.trim();
-      const match = text.match(OPENING_TAG_REGEX);
-      if (match) {
-        displayType = match[0] || ""; // Full match including < and >
-      }
-    }
-  }
```

Comment:
no need for all of that, the displayType is in the attributes passed to the component